vcd/terraform_binary_test.go

```diff
@@ -0,0 +1,148 @@
+		"CatalogItem":                  testConfig.VCD.Catalog.CatalogItem,
+		"OvaPath":                      testConfig.Ova.OvaPath,
+		"MediaPath":                    testConfig.Media.MediaPath,
+		"MediaUploadPieceSize":         testConfig.Media.UploadPieceSize,
+		"MediaUploadProgress":          testConfig.Media.UploadProgress,
+		"OvaDownloadUrl":               testConfig.Ova.OvaDownloadUrl,
+		"OvaTestFileName":              testConfig.Ova.OvaTestFileName,
+		"OvaUploadProgress":            testConfig.Ova.UploadProgress,
+		"OvaUploadPieceSize":           testConfig.Ova.UploadPieceSize,
+		"OvaPreserve":                  testConfig.Ova.Preserve,
+		"LoggingEnabled":               testConfig.Logging.Enabled,
+		"LoggingFileName":              testConfig.Logging.LogFileName,
+		"EdgeGateway":                  testConfig.Networking.EdgeGateway,
+		"SharedSecret":                 testConfig.Networking.SharedSecret,
+		"ExternalNetwork":              testConfig.Networking.ExternalNetwork,
+		"ExternalNetworkPortGroup":     testConfig.Networking.ExternalNetworkPortGroup,
+		"ExternalNetworkPortGroupType": testConfig.Networking.ExternalNetworkPortGroupType,
+		"ExternalIp":                   testConfig.Networking.ExternalIp,
+		"InternalIp":                   testConfig.Networking.InternalIp,
+		"Vcenter":                      testConfig.Networking.Vcenter,
+		"LocalIp":                      testConfig.Networking.Local.LocalIp,
+		"LocalGateway":                 testConfig.Networking.Local.LocalSubnetGateway,
+		"PeerIp":                       testConfig.Networking.Peer.PeerIp,
+		"PeerGateway":                  testConfig.Networking.Peer.PeerSubnetGateway,
+		"MaxRetryTimeout":              testConfig.Provider.MaxRetryTimeout,
+		"AllowInsecure":                testConfig.Provider.AllowInsecure,
+		"ProviderSysOrg":               testConfig.Provider.SysOrg,
+		"ProviderUrl":                  testConfig.Provider.Url,
+		"ProviderUser":                 testConfig.Provider.User,
+		"ProviderPassword":             testConfig.Provider.Password,
+		"Tags":                         "custom",
+		"Prefix":                       "cust",
+		"CallerFileName":               "",
+	}
+	for _, fileName := range binaryTestList {
+
+		baseName := strings.Replace(path.Base(fileName), ".tf", "", -1)
+
+		targetScript := path.Join(testArtifactsDirectory, params["Prefix"].(string)+"."+baseName+".tf")
+		// It the target file exists, we remove it, as we need a fresh one to be generated
+		if fileExists(targetScript) {
+			err := os.Remove(targetScript)
+			if err != nil {
+				t.Logf("error removing %s\n", targetScript)
+				panic("can't remove essential file")
+			}
+		}
+
+		sourceFile := path.Join(getCurrentDir(), customTemplatesDirectory, fileName)
+		templateText, err := readFile(sourceFile)
+		if err != nil {
+
+			t.Logf("error reading from %s: %s", fileName, err)
+		}
+		params["FuncName"] = baseName
+
+		reHasProvider := regexp.MustCompile(`(?m)^\s*provider\s+"`)
```

Comment:
What this prefix `re` means?

Reply:
`re` stands for "regular expression". I usually prefix variables that have a regular expression with this.